src/mflux/server/openai_server.py

```diff
@@ -0,0 +1,243 @@
+
+# Configure the logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s'))
+logger.addHandler(console_handler)
+
+app = FastAPI()
+
+# Initialize the model manager
+model_manager = ModelManager()
+
+class GenerationRequest(BaseModel):
+    prompt: str
+    model: Optional[str] = "schnell"
+    base_model: Optional[str] = None
+    height: Optional[int] = HEIGHT
+    width: Optional[int] = WIDTH
+    size: Optional[str] = None
+    quality: Optional[str] = None
+    guidance: Optional[float] = 4.0
+    image_strength: Optional[float] = None
+    controlnet_strength: Optional[float] = None
+    seed: Optional[int] = None
+    quantize: Optional[int] = 8
+    n: Optional[int] = 1
+    lora_repo_id: Optional[str] = None
+    lora_scale: Optional[float] = 0.7
+    low_ram: Optional[bool] = False
+    stepwise_output: Optional[bool] = False
+    metadata: Optional[bool] = False
+    local_path: Optional[str] = None
+    response_format: Optional[str] = None
+    keep_alive: Optional[int] = MODEL_UNLOAD_IMMEDIATELY
+
+class Image(BaseModel):
+    b64_json: str
+    
+class GenerationResponse(BaseModel):
+    created: int
+    data: List[Image]
+
+@app.post("/v1/images/generations", response_model=GenerationResponse)
+async def generate_images(request: GenerationRequest = Body(...)):
+    try:
+        # Determine the image path (if an image upload feature is needed, it would require a different approach)
+        image_path = None
+        
+        # Setup for stepwise output if requested
+        stepwise_dir = None
+        if request.stepwise_output:
+            stepwise_dir = create_temp_directory()
+
+        # Use the model name directly without OpenAI mapping
+        model = request.model or "schnell"
+        
+        # For OpenAI API compatibility, we can validate sizes
+        if model in ["dall-e-3", "schnell"]:
+            valid_sizes = ['1024x1024', '1792x1024', '1024x1792']
+            if model == "dall-e-3":
+                model = "schnell"
+        elif model in ['dall-e-2', 'dev']:
+            valid_sizes = ['256x256', '512x512', '1024x1024']
+            if model == "dall-e-2":
+                model = "dev"
+        else:
+            # For custom models, we'll be more flexible with sizes
+            valid_sizes = None
+        
+        # Handle size parameter
+        if request.size is not None:
+            # Check if size is valid for the selected model
+            if valid_sizes and request.size not in valid_sizes:
+                raise HTTPException(
+                    status_code=http.client.BAD_REQUEST, 
+                    detail=f"Invalid size '{request.size}' for model '{model}'. Valid sizes: {valid_sizes}"
+                )
+            width_str, height_str = request.size.lower().split('x')
```

Comment:
`.partition` is faster than `.split()` since it can only ever do up to a single split (which is what you want here, anyway):
```suggestion
            width_str, _, height_str = request.size.lower().partition('x')
```